src/typing_extensions.py

```diff
@@ -1795,28 +1795,56 @@ def __parameters__(self):
             return tuple(
                 tp for tp in self.__args__ if isinstance(tp, (typing.TypeVar, ParamSpec))
             )
+# 3.10+
+else:
+    _ConcatenateGenericAlias = typing._ConcatenateGenericAlias
 
+# 3.8-3.9.2
+class _EllipsisDummy: ...
```

Comment:
If you make this inherit from ParamSpec, can it fix 3.10.3+?

Reply:
I might be missing something here. To what case are you referring here?
This backport of `Concatenate[...]` also works for all 3.10, as the critical part is in `_concatenate_getitem`

Is it possibly related to the following test I jointly introduced also in #479?

```python
def test_alias_subscription_with_ellipsis(self):
    P = ParamSpec('P')
    X = Callable[Concatenate[int, P], Any]
    C1 = X[...]  # <-- Needs PR #479 for 3.10.3+
    self.assertEqual(get_args(C1), (Concatenate[int, ...], Any))  # <-- needs this PR
```

---

That aside; It is possible to do `_EllipsisDummy = ParamSpec("_EllipsisDummy")` if needed as subclassing `ParamSpec` is (currently) not possible. I thought using a new class might have less side-effects than using a ParamSpec instance.

Reply:
I'm referring to the comment right above here, about the test that works only in 3.10.0-3.10.2 but not 3.10.3+. Making `_EllipsisDummy` an instance of ParamSpec might fix that test for later versions of 3.10.

Reply:
I am sorry for the confusion. I do think we talk about the same location :), if not correct me please. I made the necessary edits above.

#479 allows to backport that test to all versions of 3.10, however for #479 the test `self.assertEqual(get_args(C1), (Concatenate[int, ...], Any))` would need this PR to work without an error on 3.10 as well.  
Both PRs are needed make this test work for the 3.10 version range.

Reply:
I see, thanks. #479 is merged now, so would you mind updating this PR? There's a merge conflict right now.

Reply:
Sure. Unified the code and the test we talked about, aside from that added one more test for invalid usage.